src/main/java/com/sayweb/number/SayNumber.java

```diff
@@ -21,41 +23,76 @@ public class SayNumber {
     "eighteen",
     "nineteen"
   };
-  private static final String[] tens = {
+
+  private static final String[] TENS = {
     "", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"
   };
 
+  // Cache for word-to-number mapping to avoid regenerating
+  public static final Map<Long, String> NUMBER_TO_WORDS_CACHE = new HashMap<>();
```

Comment:
1. NUMBER_TO_WORDS_CACHE should be private and be accessed via methods only. (To limit putting unintended  values into it, outside this class in future.)

2. Set its storage size limited. Use some collection deleting elements after some time or the oldest ones. (Otherwise, our collection might keep obsolete elements for several years, leading our app eating all the memory on a server.)